src/ic-cdk/src/api/call.rs

```diff
@@ -57,6 +57,11 @@ pub type CallResult<R> = Result<R, (RejectionCode, String)>;
 struct CallFutureState {
     result: Option<CallResult<Vec<u8>>>,
     waker: Option<Waker>,
+    id: Principal,
+    method: String,
+    arg: Vec<u8>,
+    payment: u64,
+    payment128: u128,
```

Comment:
If we're removing the closure, you may as well remove the 64-bit code altogether. Just store a u128 here and use add_payment for both.